lua/starfall/libs_sh/vmatrix.lua

```diff
@@ -65,116 +65,184 @@ end
+	m:Scale(vec)
 end
 
 --- Scales the absolute translation
 -- @param num Amount to scale by
-function vmatrix_methods:scaleTranslation (num)
+function vmatrix_methods:scaleTranslation(num)
 	SF.CheckLuaType(num, TYPE_NUMBER)
 
-	local v = unwrap(self)
-	v:ScaleTranslation(num)
+	local m = unwrap(self)
+	m:ScaleTranslation(num)
 end
 
 --- Sets the angles
 -- @param ang New angles
-function vmatrix_methods:setAngles (ang)
+function vmatrix_methods:setAngles(ang)
 	SF.CheckType(ang, ang_meta)
 
-	local v = unwrap(self)
-	v:SetAngles(SF.UnwrapObject(ang))
+	local m = unwrap(self)
+	m:SetAngles(SF.UnwrapObject(ang))
 end
 
 --- Sets the translation
 -- @param vec New translation
-function vmatrix_methods:setTranslation (vec)
+function vmatrix_methods:setTranslation(vec)
 	SF.CheckType(vec, vec_meta)
 	local vec = vunwrap(vec)
 
-	local v = unwrap(self)
-	v:SetTranslation(vec)
+	local m = unwrap(self)
+	m:SetTranslation(vec)
+end
+
+--- Sets the forward direction of the matrix. First column
+-- @param forward The forward vector
+function vmatrix_methods:setForward(forward)
+	SF.CheckType(forward, vec_meta)
+	
+	local m = unwrap(self)
+	m:SetForward(vunwrap(forward))
+end
+
+--- Sets the right direction of the matrix. Negated second column
+-- @param right The right vector
+function vmatrix_methods:setRight(right)
+	SF.CheckType(right, vec_meta)
+	
+	local m = unwrap(self)
+	m:SetRight(vunwrap(right))
+end
+
+--- Sets the up direction of the matrix. Third column
+-- @param up The up vector
+function vmatrix_methods:setUp(up)
+	SF.CheckType(up, vec_meta)
+	
+	local m = unwrap(self)
+	m:SetUp(vunwrap(up))
 end
 
 --- Sets a specific field in the matrix
 -- @param row A number from 1 to 4
 -- @param column A number from 1 to 4
 -- @param value Value to set
 function vmatrix_methods:setField(row, column, value)
-	local v = unwrap(self)
-	v:SetField(row, column, value)
+	local m = unwrap(self)
+	m:SetField(row, column, value)
+end
+
+--- Copies the values from the second matrix to the first matrix. Self-Modifies
+-- @param src Second matrix
+function vmatrix_methods:set(src)
+	SF.CheckType(src, vmatrix_metamethods)
+
+	local m = unwrap(self)
+	m:Set(unwrap(src))
+end
+
+--- Initializes the matrix as Identity matrix
+function vmatrix_methods:setIdentity()
+	local m = unwrap(self)
+	m:Identity()
+end
+
+--- Returns whether the matrix is equal to Identity matrix or not
+-- @return bool True/False
+function vmatrix_methods:isIdentity()
+	local m = unwrap(self)
+	return m:IsIdentity()
+end
+
+--- Returns whether the matrix is a rotation matrix or not. Checks if the forward, right and up vectors are orthogonal and normalized.
+-- @return bool True/False
+function vmatrix_methods:isRotationMatrix()
+	local m = unwrap(self)
+	return m:IsRotationMatrix()
+end
+
+--- Inverts the matrix. Inverting the matrix will fail if its determinant is 0 or close to 0
+-- @return bool Whether the matrix was inverted or not
+function vmatrix_methods:invert()
```

Comment:
I think it should be isInverted() because invert() suggests that it does inversion.

Reply:
It does inversion, but if the matrix determimant is 0 then it is not possible to invert matrix, so boolean returned on success or fail
http://wiki.garrysmod.com/page/VMatrix/Invert